transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java

```diff
@@ -270,6 +270,33 @@ public boolean remove() {
     }
 
     /**
+     * Removes the current flushed message and returns its {@link ChannelPromise}. Unlike {@link #remove()} this method
+     * does not release the message or complete the promise. If no flushed message exist, this method returns
+     * {@code null}.
+     */
+    public ChannelPromise steal() {
```

Comment:
@buchgr I dont like adding this method too much... Let me think about it a bit but maybe its ok to merge this first.